sklearn/cluster/tests/test_affinity_propagation.py

```diff
@@ -56,8 +57,21 @@ def test_affinity_propagation():
                                              copy=False)
     assert_array_equal(labels, labels_no_copy)
 
+    # Test preference string input
+    preference = '100%'
+    af = AffinityPropagation(preference=preference, verbose=True)
+    labels = af.fit(X).labels_
+    assert_equal(max(labels), n_samples-1)
+
+    preference = '0%'
+    af = AffinityPropagation(preference=preference, verbose=True)
+    labels = af.fit(X).labels_
+    assert_equal(max(labels), 2)
+
     # Test input validation
     assert_raises(ValueError, affinity_propagation, S[:, :-1])
+    assert_raises(ValueError, affinity_propagation, S, preference='-101%')
+    assert_raises(ValueError, affinity_propagation, S, preference='101.3%')
```

Comment:
Please also check that the error message is appropriate. Check that a string representation of a float is also rejected